src/solvers/flattening/arrays.h

```diff
@@ -77,6 +82,27 @@ class arrayst:public equalityt
   typedef std::map<std::size_t, index_sett> index_mapt;
   index_mapt index_map;
 
+  enum class constraint_typet
+  {
+    ARRAY_ACKERMANN,
+    ARRAY_EQUALITY,
+    ARRAY_WITH,
+    ARRAY_WITH_OTHER,
+    ARRAY_IF_TRUE,
+    ARRAY_IF_FALSE,
+    ARRAY_OF,
+    ARRAY_TYPECAST,
+    ARRAY_CONSTANT,
+    ARRAY_NON_CONSTANT,
+    ARRAY_COMPREHENSION
+  };
+
+  typedef std::list<exprt> array_constraintst;
+  typedef std::map<constraint_typet, array_constraintst> array_constraints_mapt;
+  array_constraints_mapt array_constraints_map;
+  void display_constraints();
+  std::string enum_to_string(constraint_typet);
```

Comment:
Is there a better way of doing this? Wanted to include more specific information regarding the type of constraint. For example split constraints added for array_with to constraints added for the specific index and for other indices.